docs/advanced/action-handlers.md

```diff
@@ -1,19 +1,28 @@
 # Action Handlers
+
 Event sourcing involves modeling the state changes made by applications as an immutable sequence or “log” of events.
 Instead of focussing on current state, you focus on the changes that have occurred over time. It is the practice of
 modelling your system as a sequence of events. In NGXS, we called this Action Handlers.
 
 Typically actions directly correspond to state changes but it can be difficult to always make your component react
-based on state. As a side effect of this paradigm, we end up creating lots of intermediate state properrties
+based on state. As a side effect of this paradigm, we end up creating lots of intermediate state properties
 to do things like reset a form/etc. Action handlers let us drive our components based on state along with events
 that emit.
 
 For example, if we were to have a shopping cart and we were to delete an item out of it you might want to show
 a notification that it was successfully removed. In a pure state driven application, you might create some kind
-of message array to make the dialog show up. With Action Handlers, we can respond to the action directly. 
+of message array to make the dialog show up. With Action Handlers, we can respond to the action directly.
 
 The action handler is a observable that recieves all the actions dispatched before the state takes any action on it.
-Since its an observable, we can use pipes and we created a `ofAction` pipe to make filtering the actions easier.
+
+Actions in NGXS also have a lifecycle. Since any potential action can be async we tag actions showing whether they are "DISPATCHED" or "COMPLETED". This gives you the ability to react to actions at different points in their existence.
+
+Since its an observable, we can use pipes and created 3 of them.
+
+* `ofAction`: triggers both when an action has been dispatched and when it completes
+* `ofActionDispatched`: triggers when an action has been dispatched but NOT when it completes
+* `ofActionCompleted`: triggers when an action has been completed but NOT when it is dispatched
```

Comment:
We might want a `ofActionErrored` too. Just a thought.

Reply:
Ill look into it